### PR TITLE
Prevent selectBox popup opening after clear button click (T680755)

### LINKED ISSUE #3995: dxSelectBox, dxTagBox - When searching for non-existing value popup with 'No data to display' blinks and hides
### Prerequisites:
- list of items
- search enabled

### Steps to Reproduce:

1. Click on dropdown, see the list of items appears.
2. Type something that doesn't match with list items (e.g. qqqq) for search.
3. See the popup with 'No data to display' appears and immediately disappears, sometimes it's even impossible to read the wording.

### Reproduction:
Plunker: http://plnkr.co/edit/5mkoG2v7zV2eprmDbbJh?p=preview

### Results You Received:
The popup with 'No data to display' appears and immediately disappears

### Results You Expected:
When the popup with 'No data to display' appears it shouldn't immediately disappear. Or ideally it should be configurable.

### Environment Details:
devexteme version: 17.2.7
devextreme-angular version: 17.2.7
Chrome 66, Windows 10 x64 (1920x1080 resolution)
It's reproducible on all platforms that Devextreme provides (jQuery, AngularJS, Angular, ASP.NET MVC, ASP.NET Core).

### Issue in devextreme-angular repo:
https://github.com/DevExpress/devextreme-angular/issues/772

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**
Resolve #3995

### CODE REVIEW

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]
This behavior looks like incorrect:

* DropDownLists (dxSelectBox, dxTagBox etc) should be closed after clear button clicked
* DropDownEditors (dxDropDownBox, dxColorBox) should not be closed after clear button clicked
* All DropDownEditors should not be closed after the value was changed to `null` using API without clear button click.

I think this is incorrect. Please correct the pull request or explain why described behavior is correct

File: js/ui/drop_down_editor/ui.drop_down_list.js [hunk 1]
```diff
@@ -730,12 +730,16 @@ var DropDownList = DropDownEditor.inherit({
         this._refreshPopupVisibility();
     },
 
+    _needToOpenPopup: function() {
```
Comment (mrjohn0011) [inline js/ui/drop_down_editor/ui.drop_down_list.js]: Use `_needToOpenPopup: this._hasItemsToShow.bind(this)`

File: js/ui/select_box.js [hunk 2]
```diff
@@ -624,6 +624,10 @@ var SelectBox = DropDownList.inherit({
         this.option("value", null);
     },
 
+    _needToOpenPopup: function() {
```
Comment (mrjohn0011) [inline js/ui/select_box.js]: `_needToOpenPopup: this._needPassDataSourceToList.bind(this)`

**Review by mrjohn0011 — COMMENTED** [review]
Some refactoring is required

File: js/ui/drop_down_editor/ui.drop_down_list.js (hunk 1 above)
Comment (DokaRus) [reply js/ui/drop_down_editor/ui.drop_down_list.js]: I suggest to rename it to something like `_isNeedToOpenPopup`

**Review by mrjohn0011 — DISMISSED** [review]

**Review by mrjohn0011 — APPROVED** [review]

**Pull request merged**